### PR TITLE
Refactor/configure students, student, and profile page components, controllers, and routes

### LINKED ISSUE #74: Refactor nomenclature to create student page
Components:
- StudentPage
- StudentGrid
- StudentCard

### LINKED ISSUE #75: Create students index page components
Components:
- StudentsPage
- StudentsGrid

### LINKED ISSUE #76: Create students show and index routes and actions

### LINKED ISSUE #78: Refactor account components to profile components
Components:
- AccountPage => ProfilePage
- AccountCard (parent of AccountPreview and AccountOptions)
- ProfileMetadata => AccountPreview
- ProfileInfo => AccountOptions

### PR DESCRIPTION
**Pull request by warrenshen**
Fix #74 
Fix #75 
Fix #76 
Fix #78 

Featured additions/changes:

Routes:
- /students (index)
- /students/:id (show)
- /profile

Students index view:
- StudentsPage
- StudentsFilters
- StudentsGrid

Student show view:
- ProfilePage => StudentPage
- ProfileCardGrid => StudentGrid
- ProfileCard => StudentCard

Profile view:
- AccountPage => ProfilePage
- AccountCard (parent of AccountPreview and AccountOptions)
- ProfileMetadata => AccountPreview
- ProfileInfo => AccountOptions

### CODE REVIEW

File: app/assets/javascripts/components/header.es6.jsx [hunk 1]
```diff
@@ -2,13 +2,13 @@ class Header extends Component {
 
   static get propTypes() {
     return {
-      handleSidebarClick: React.PropTypes.func.isRequired,
+      handleClick: React.PropTypes.func.isRequired,
```
Comment (warrenshen) [inline app/assets/javascripts/components/header.es6.jsx]: I just didn't like the func name `handleSidebarClick` so for now we're defaulting to a more general name.

File: app/assets/javascripts/components/profile_cards.es6.jsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class ProfileCards extends Component {
```
Comment (warrenshen) [inline app/assets/javascripts/components/profile_cards.es6.jsx]: @sonia-y this is the parent component of `ProfilePreview` and `ProfileOptions` I was talking about!

File: app/assets/javascripts/components/sidebar_group.es6.jsx [hunk 3]
```diff
@@ -10,16 +10,16 @@ class SidebarGroup extends React.Component {
   }
 
   render() {
-    return(
+    return (
```
Comment (unzunz) [inline app/assets/javascripts/components/sidebar_group.es6.jsx]: space after return!

File: app/assets/javascripts/components/profile_cards.es6.jsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  get styles() {
+    return {
+      container: {
+        display: 'flex', 
+        flexDirection: 'column',
+        flex: 1,
+        position: 'relative',
+        padding: '0 12px',
```
Comment (maxwolffe) [inline app/assets/javascripts/components/profile_cards.es6.jsx]: padding: '0px 12px' instead?

**Comment by maxwolffe** [comment]
This is a relatively phat PR. It might have been better to break this into two smaller PRs to make it easier to review. Otherwise, LGTM!

**Pull request merged**